components/org.wso2.carbon.deployment.engine/src/test/java/org/wso2/carbon/deployment/engine/deployers/CustomDeployer.java

```diff
@@ -93,16 +105,24 @@ public void undeploy(Object key) throws CarbonDeploymentException {
                     "is not a String value");
         }
         logger.info("Undeploying : " + key);
-        try {
-            File fileToUndeploy = new File(testDir + File.separator + key);
-            logger.info("File to undeploy : " + fileToUndeploy.getAbsolutePath());
-            FileInputStream fis = new FileInputStream(fileToUndeploy);
+        File fileToUndeploy;
+        if (key.equals("sample1.txt")) {
```

Comment:
Shall we turn-around the equal check for avoiding any NPEs ?
For ex: "something".equals(string)